pysteps/blending/steps.py

```diff
@@ -1440,19 +1440,27 @@ def calculate_weights_spn(correlations, cov):
             raise ValueError("cov must contain a covariance matrix")
         else:
             # Make a numpy matrix out of cov and get the inverse
+            cov = np.where(cov == 0.0, 10e-5, cov)
+            if cov[0][0] != 1.0:
+                cov[0][0] = 1.0
+            if cov[1][1] != 1.0:
+                cov[1][1] = 1.0
             cov_matrix = np.asmatrix(cov)
             cov_matrix_inv = cov_matrix.getI()
```

Comment:
do we need to capture LinAlgError in case of a singular matrix or is this dealt somewhere else?

Reply:
I'll double check that, good point.

Reply:
is this now solved by 27ed33a?